openmp/libompd/src/omp-icv.cpp

```diff
@@ -791,13 +796,73 @@ ompd_get_schedule(ompd_task_handle_t *task_handle, /* IN: OpenMP task handle*/
   ompd_rc_t ret = sched
                       .access("r_sched_type") // td->td_icvs.sched.r_sched_type
                       .castBase()
-                      .getValue(*kind);
-  if (ret != ompd_rc_ok)
+                      .getValue(kind);
+  if (ret != ompd_rc_ok) {
     return ret;
-  ret = sched
-            .access("chunk") // td->td_icvs.sched.r_sched_type
-            .castBase()
-            .getValue(*modifier);
+  }
+  char *run_sched_var_string;
+  ret = callbacks->alloc_memory(100, (void **) &run_sched_var_string);
+  if (ret != ompd_rc_ok) {
+    return ret;
+  }
+  run_sched_var_string[0] = '\0';
+  if (SCHEDULE_HAS_MONOTONIC(kind)) {
+    strcpy(run_sched_var_string, "monotonic:");
+  } else if (SCHEDULE_HAS_NONMONOTONIC(kind)) {
+    strcpy(run_sched_var_string, "nonmonotonic:");
+  }
+
+  bool static_unchunked = false;
+  switch (SCHEDULE_WITHOUT_MODIFIERS(kind)) {
+    case kmp_sch_static:
+    case kmp_sch_static_greedy:
+    case kmp_sch_static_balanced:
+      static_unchunked = true;
+      strcat(run_sched_var_string, "static");
+      break;
+    case kmp_sch_static_chunked:
+      strcat(run_sched_var_string, "static");
+      break;
+    case kmp_sch_dynamic_chunked:
+      strcat(run_sched_var_string, "dynamic");
+      break;
+    case kmp_sch_guided_chunked:
+    case kmp_sch_guided_iterative_chunked:
+    case kmp_sch_guided_analytical_chunked:
+      strcat(run_sched_var_string, "guided");
+      break;
+    case kmp_sch_auto:
+      strcat(run_sched_var_string, "auto");
+      break;
+    case kmp_sch_trapezoidal:
+      strcat(run_sched_var_string, "trapezoidal");
+      break;
+    case kmp_sch_static_steal:
+      strcat(run_sched_var_string, "static_steal");
+      break;
+    default:
+      ret = callbacks->free_memory((void *)(run_sched_var_string));
+      if (ret != ompd_rc_ok) {
+        return ret;
+      }
+      ret = create_empty_string(run_sched_string);
+      return ret;
+  }
+
+  int chunk = 0;
+  if (static_unchunked == false){
+    ret = sched
+          .access("chunk") // td->td_icvs.sched.chunk
+          .castBase()
+          .getValue(chunk);
+    if (ret != ompd_rc_ok) {
+      return ret;
+    }
+  }
```

Comment:
In case of an error, we will leak the allocation for the string. I suggest to move this up before the allocation.
I think it's in general better to collect all data before allocating and formating the string.

Reply:
Done. Also added a comment wrt setting chunk to 0 (in the case of static scheduling where chunk is not specified) to be in sync with what OMPT returns (to avoid test failures when we compare OMPD-OMPT data). Also added a semicolon which was missed out in the warnings PR. (#39 )